src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
+  [].__proto__.join2 = function(separator = ',') {
     // write code here
+    let str = '';
+
+    for (let i = 0; i < this.length; i++) {
+      let tmp = this[i];
+
+      if (tmp === undefined || tmp === null) {
```

Comment:
Replace `tmp === undefined || tmp === null` with `!tmp`

Reply:
if  chenge !tmp
undefined = null = NaN = 0

test(`Should work when array has null as a first element`, () => {
  expect([null, null, 1, undefined, 0, NaN, ''].join2(','))
    .toBe(',,1,,0,NaN,');    result=  .toBe(',,1,,,,');